core/types/scenario.go

```diff
@@ -167,6 +176,27 @@ func (si *ScenarioItem) validate() error {
 	return nil
 }
 
+func (si *ScenarioItem) ParseTLC(certFile, keyFile string) error {
```

Comment:
**_Nice to have;_**

What about changing the contract to below;
`func ParseTLC(certFile, keyFile string) (string, string, error)`

We can return `(cert, certpool, err)` if suitable, `(nil, nil, err)` if not suitable. Then we don't need the `if` statements both on `main.go` and` json.go`:
https://github.com/zhiburt/ddosify/blob/patch-tls-certs/main.go#L252
https://github.com/zhiburt/ddosify/blob/patch-tls-certs/config/json.go#L151

 WDYT?

Reply:
Well... you will then ignore error?
Or you will have to introduce a custom error type to recognize if it failed because the path was empty or there was some issue.

I agree that the interface of the function maybe not the best (see how it used it tests :disappointed_relieved:).
But what I am saying the check will not go away right (because of what I've pointed out above)?

Reply:
No need to discard error logic. Please take a look `AdjustUrlProtocol`. We can apply the same strategy. As a said it is nice to have for now.

Reply:
https://github.com/ddosify/ddosify/blob/3db5d92513e426998badb619ecc783135d84584e/core/types/scenario.go#L179